File: frontend-app/src/@types/TodoItem.ts
export type TodoItem = {
  id: number;
  title: string;
  description: string;
  assignedToId: string;
  isCompleted: boolean;
}

export type UpdateTodoItemModel = {
  markComplete: boolean;
  assignedToId: string;
}

export type CreateTodoItemRequest = {
  title: string;
  description: string;
  assignedTo?: string;
}

File: services/project-api/src/Nexus.Project.Api/Entities/TodoItem.cs
using Nexus.Common;

namespace Nexus.ProjectAPI.Entities;

public class TodoItem : AuditableNexusEntityBase
{
    public required string Title { get; set; }

    public string Description { get; set; } = string.Empty;

    public string? AssignedToId { get; set; }

    public bool IsCompleted { get; private set; }
    
    public int ProjectId { get; set; }

    public void MarkComplete(bool isCompleted)
    {
        IsCompleted = isCompleted;
    }

    public void AssignTodoItem(string assignedToId)
    {
        AssignedToId = assignedToId;
    }
}